src/backend/commands/tablecmds.c

```diff
@@ -3997,7 +3997,7 @@ RenameRelation(RenameStmt *stmt)
 	if (gp_allow_rename_relation_without_lock)
 		targetrelation = fake_relation_open(relid);
 	else
-		targetrelation = relation_open(relid, AccessExclusiveLock);
```

Comment:
Will main branch has the same issue here?

Reply:
Yes, it also havs this issue.

Reply:
> Yes, it also has this issue.

@Aegeaner So let's first open a PR to main branch and after main branch pushed then cherry-pick to merge branch.

Plus, what issue for this issue? why there is no change of case?

Reply:
This is because in the GPDB-specific code in `RenameRelation`, it used `AccessExclusiveLock` unconditionally to open the target relation, but for index statement, it should use `ShareUpdateExclusiveLock`. After this fix, it aligned with upstream, so there is no change in test case.

Before this change, there was a diff in the test case:
```
diff -I HINT: -I CONTEXT: -I GP_IGNORE: -U3 /home/gpadmin/gpdb-postgres-merge/src/test/regress/expected/alter_table.out /home/gpadmin/gpdb-postgres-merge/src/test/regress/results/alter_table.out
--- /home/gpadmin/gpdb-postgres-merge/src/test/regress/expected/alter_table.out 2022-12-14 13:17:22.965202903 +0800
+++ /home/gpadmin/gpdb-postgres-merge/src/test/regress/results/alter_table.out  2022-12-14 13:17:23.089203851 +0800
@@ -260,9 +260,11 @@
 ORDER BY relation::regclass::text COLLATE "C";
               relation              |           mode
 ------------------------------------+--------------------------
+ alter_idx_rename_test_idx_2        | AccessExclusiveLock
  alter_idx_rename_test_idx_2        | ShareUpdateExclusiveLock
+ alter_idx_rename_test_parted_idx_2 | AccessExclusiveLock
  alter_idx_rename_test_parted_idx_2 | ShareUpdateExclusiveLock
-(2 rows)
+(4 rows) COMMIT;
 BEGIN;
```

Reply:
Let's open a PR to main branch for this commit and let others review, thanks.

Reply:
Should I copy the related test case (See above) to the main branch PR too?

Reply:
@kainwen Perhaps we don't need to open a PR to main branch, since this lock relaxing improvement was only introduced in PG 12, by the following two commits:

[Lower lock level for renaming indexes ](https://github.com/greenplum-db/gpdb-postgres-merge/commit/1b5d797cd4f7133ff0d18e123fcf41c67a5a7b0b)

[Ensure correct lock level is used in ALTER ... RENAME ](
https://github.com/greenplum-db/gpdb-postgres-merge/commit/6c8d1c197bd67e78c12c68d8f0bd111a59e7bccd)

Reply:
`RelationReloadIndexInfo()` has lower the lock level needed to reload relcache for index relations to `AccessShareLock`, which provided the possibility for relaxing lock level for index statements, but for update correctness, it still used `ShareUpdateExclusiveLock` lockmode for indexes renaming statements.

Reply:
The main branch has the [Lower lock level for renaming indexes](https://github.com/greenplum-db/gpdb-postgres-merge/commit/1b5d797cd4f7133ff0d18e123fcf41c67a5a7b0b) and it grabs an `AccessExclusiveLock` for `GpRenameChildPartitions()`.

I think the main branch needs a PR.